app/Http/Controllers/Api/ServerController.php

```diff
@@ -102,7 +102,53 @@ public function edit(Request $request, int $serverId)
 
         return response()->json(['message' => 'Server updated successfully.']);
     }
+    public function showSettings($serverId)
+    {
+        $server = Server::find($serverId);
+        if (!$server) {
+            return response()->json(['message' => 'Server not found.'], 404);
+        }
+    
+        return Inertia::render('Settings/Server', [
+            'server' => $server,
+        ]);
+    }
+    public function destroy(Request $request, int $serverId)
+{
+    $server = Server::find($serverId);
+
+    if (!$server) {
+        return response()->json(['message' => 'Server not found.'], 404);
+    }
+
+    $server->users()->detach();
+    $server->delete();
+
+    return redirect('/home');
+}
+public function update(Request $request, int $serverId)
+{
+    $request->validate([
+        'name' => 'nullable|string|max:50',
+        'description'=> 'nullable|string|max:255',
+        'icon' => 'nullable|image|mimes:jpeg,png,jpg|max:2048',
+    ]);
+
+    $server = Server::find($serverId);
+    if (!$server) return redirect()->back()->withErrors(['message' => 'Server not found.']);
+
+    if ($request->file('icon')?->isValid()) {
+        $path = $request->file('icon')->store('uploads', 'public');
+        $server->icon = Storage::url($path);
+    }
+
+    $server->name = $request->input('name', $server->name);
+    $server->description = $request->input('description', $server->description);
+    $server->save();
 
+    return redirect()->route('settings.server', ['serverId' => $serverId])
+        ->with('message', 'Server updated successfully.');
+}
```

Comment:
While you didnt do nothing wrong there are two thing that i wanted to say, 1) Ask about what route we have. 2)We already had routes for editing and deleting, and while i understand that it easier to just redirect here, it add problems with consistence and testing.